docs/developer-docs/backend/motoko/0.11.0-migration-guide.mdx

```diff
@@ -0,0 +1,90 @@
+---
+keywords: [motoko, 0.11.0, migration guide, safety feature, breaking change]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Motoko v0.11.0: Safety feature migration guide
+
+<MarkdownChipRow labels={["Motoko", "Migration guide"]} />
+
+## Overview
+
+Motoko v0.11.0 and newer introduces a new safety feature that could cause breaking changes to existing workflows.
+
+In previous Motoko versions, third-party library functions could make calls to sensitive functions such as `ExperimentalCycles.add(...)` and `Timer.setTimer(...)` without providing indication that to the caller that sensitive system functionality was being used.
+
+In Motoko 0.11.0 and newer, the type system is used to detect and prevent this scenario while still allowing developers to grant access to sensitive functions.
+
+## Motoko v0.11.0 changes
+
+Motoko uses a simple capability-based type system which selectively enables language constructs depending on their context. One example is only `async` expressions can contain `await` expressions, or only some asynchronous contexts can call shared functions.
+
+In Motoko v0.11.0 and newer, this capability system is expanded to restrict calls to functions including `Timer.setTimer` and `ExperimentalCycles.add`. To accomplish this, Motoko introduces and uses the pseudo-type parameter `system` and corresponding pseudo-type argument (also system) for both functions and classes.
+
+Another change in v0.11.0 and newer is a revision of `ExperimentalCycles.add`, which has been changed from `Nat -> ()` to `<system>Nat -> ()`. This change supports the system capability requirement of the special type parameter `<system>`.
+
+:::info
+The `system` type parameter must be the first parameter of any function, class constructor or function type, e.g. `<system, T, U>` is valid but `<T, system, U>` is not.
+:::
+
+## Migrations for existing workflows
+
+User-defined functions must now declare an explicit system pseudo-type parameter if they require system capabilities. If a caller wants to grant system capabilities to a callee, the callee must have system capabilities either through the program's context or the call must reside within a function or class that requires a system pseudo parameter.
```

Comment:
```suggestion
User-defined functions must now declare an explicit system pseudo-type parameter if they require system capabilities. If a caller wants to grant system capabilities to a callee, the caller must already have system capabilities either (implicitly) by virtue of the callsite's progam context or (explicitly) because the callsite resides within a function or class that itself declares the new `system`  type parameter.
```

Reply:
```suggestion
User-defined functions must now declare an explicit system pseudo-type parameter if they require system capabilities. If a caller wants to grant system capabilities to a callee, the caller must have system capabilities either through the program's context or the call must reside within a function or class that requires a system pseudo parameter.
```
If that's too radical, do still change callee to caller as indicated.